src/parsing/css-style-value-parsing.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//     You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//     See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(internal) {
+
+  var parsers = {};
+  parsers[CSSNumberValue] = internal.parsing.consumeNumber;
+  parsers[CSSLengthValue] = internal.parsing.consumeLength;
+  parsers[CSSTransformValue] = internal.parsing.consumeTransform;
+
+  function consume(consumer, string) {
+    if (!consumer) {
+      return null;
+    }
+    return consumer(string);
+  }
+
+  function consumeKeyword(property, string) {
+    var consumed = internal.parsing.consumeToken(/^[a-z_]+[a-z0-9_-]*/i, string);
+    if (!consumed) {
+      return null;
+    }
+    if (!internal.propertyDictionary().isValidKeyword(property, consumed[0])) {
+      return null;
+    }
+    return [new CSSKeywordValue(consumed[0]), consumed[1]];
+  }
+
+  function consumeStyleValue(property, string) {
+    if (!internal.propertyDictionary().isSupportedProperty(property)) {
+      // TODO: How do custom properties play into this?
+      return null;
+    }
+
+    var consumedKeyword = consumeKeyword(property, string);
```

Comment:
I'm assuming keywords are more common than StyleValues here. I don't really know how to determine the actual frequency of keyword vs StyleValue, but if we decided that it was more likely to be a StyleValue, we can change the order.